seismicpro/gather/gather.py

```diff
@@ -902,30 +903,21 @@ def apply_nmo(self, stacking_velocity):
 
     @batch_method(target="for")
     def sort(self, by):
-        """Sort gather `headers` and traces by specified header column.
+        """Sort gather by specified headers.
 
         Parameters
         ----------
-        by : str
-            `headers` column name to sort the gather by.
+        by : str or iterable of str
+            Headers names to sort the gather by.
 
         Returns
         -------
         self : Gather
-            Gather sorted by `by` column. Sets `sort_by` attribute to `by`.
-
-        Raises
-        ------
-        TypeError
-            If `by` is not str.
-        ValueError
-            If `by` column was not loaded in `headers`.
+            Gather sorted by given headers. Sets `sort_by` attribute to `by`.
         """
-        if not isinstance(by, str):
-            raise TypeError(f'`by` should be str, not {type(by)}')
         if self.sort_by == by:
             return self
-        order = np.argsort(self[by], kind='stable')
+        order = np.lexsort(self[to_list(by)].T[::-1])
```

Comment:
```suggestion
        order = np.lexsort(self[to_list(by)[::-1]].T)
```
For me such order is more clear, but up to you